aten/src/ATen/ParallelNative.cpp

```diff
@@ -87,15 +86,7 @@ void _run_with_pool(const std::function<void(int, size_t)>& fn, size_t range) {
   // Run the first task on the current thread directly.
   fn(0, 0);
 #else
-  caffe2::ThreadPool* pool = caffe2::mobile_threadpool();
-  if (pool) {
-    // caffe2::ThreadPool can utilize the current thread.
-    pool->run(fn, range);
-  } else {
-    for (size_t i = 0; i < range; ++i) {
-      fn(0, i);
-    }
-  }
+  native::mobile::internal::threadpool().run(fn, range);
```

Comment:
I think there is an internal API to query thread-ID: get_thread_num() https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Parallel.h#L30
Do you refer to it or some other API?

Here is one sample call-site of get_thread_num() API in aten: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/TensorIteratorReduce.cpp#L34
I think it is used to perform some ad-hoc reduction - not sure if it's actually needed by any mobile model for real.

I remember right now we are using a thread-local to mimic this behavior - I'm not sure it needs depend on specific thread pool implementation:
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/ParallelNative.cpp#L105

Reply:
Took a closer took, seems you refer to the first param (`thread_pool_task_id`) of the `fn` callback of `_run_with_pool` - I think this callback is only used in this file by `_parallel_run`, the first param is marked as `unused` there: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/ParallelNative.cpp#L135

If we move to using pthreadpool which doesn't pass thread-id to `fn` then we can simply remove it. It's doesn't seem to be in the public API of `parallel_for` callback signature (which handles task_id instead of thread_id).